modules/admin-ui/src/main/webapp/scripts/shared/services/wizards/new-event/source.js

```diff
@@ -512,7 +512,14 @@ angular.module('adminNg.services')
             minute: parseInt(endDateTime.format('mm'))
           };
 
-          defaults.presentableWeekdays = chosenSlot.format('dd');
+
+          if (!orgProperties.hasOwnProperty('admin.event.new.preselect_day')
+            || orgProperties.hasOwnProperty('admin.event.new.preselect_day')
```

Comment:
Please note that the boolean expression goes on, and since the following AND binds stronger than the preceding OR the entire expression is not always ```true```. One could, however, drop the second check for the wanted property because the second part of the OR clause is only evaluated if the existence of the wanted property is already established.

Reply:
I'm not talking about the whole condition but about this line.

Assuming `orgProperties.hasOwnProperty('admin.event.new.preselect_day')` evaluates to `false`, you end up with `if (true || …) {…}` at which point the second part of the condition does not matter any longer. The whole thing will evaluate to `true`.

Assuming `orgProperties.hasOwnProperty('admin.event.new.preselect_day')` evaluates to `true`, you end up with `if (false || true && …) {…}`. Hence, if it's necessary for the evaluation of the whole condition to evaluate the second part, the `orgProperties.hasOwnProperty('admin.event.new.preselect_day')` is always `true` making it completely unnecessary.

This makes your statement identical to:

```js
if (!orgProperties.hasOwnProperty('admin.event.new.preselect_day')
    || orgProperties['admin.event.new.preselect_day'] === true) {
```

Reply:
I'm sorry, I thought you were talking about the whole boolean expression. Of course, the second check is obsolete.